src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -643,7 +643,7 @@ CacheDirectoryElement(const char *filename)
 	appendStringInfo(directoryPath, "base/%s/", PG_JOB_CACHE_DIR);
 
 	directoryPathFound = strstr(filename, directoryPath->data);
-	if (directoryPathFound != NULL)
+	if (directoryPathFound == filename)
```

Comment:
this confirms that the `base/pgsql_job_cache/` appears at the start of the string, rather than anywhere in the string

Reply:
I would have liked it better if this comment in the code